src/instance/minecraft/chat-manager.ts

```diff
@@ -123,4 +129,35 @@ export default class ChatManager extends EventHandler<MinecraftInstance> {
       message
     })
   }
+
+  proceedFiltering(author: string, content: string): string {
+    let filteredMessage: string
+    try {
+      filteredMessage = this.clientInstance.app.profanityFilter.clean(content)
+    } catch {
+      /*
+        profanity package has bug.
+        will throw error if given one special character.
+        example: clean("?")
+        message is clear if thrown
+      */
+      filteredMessage = content
+    }
+
+    if (content !== filteredMessage) {
+      this.clientInstance.app.emit('event', {
+        localEvent: true,
+        instanceType: InstanceType.DISCORD,
```

Comment:
I assume you haven't read the documentation regarding this event.
`instanceType` is where the event has come from. Not the target. In this scenario, it came from `InstanceType.MINECRAFT`.